android/titanium/src/java/org/appcelerator/kroll/KrollDict.java

```diff
@@ -182,6 +188,22 @@ public KrollDict getKrollDict(String key)
 		}
 	}
 
+	public KrollDict[] getKrollDictArray(String key)
+	{
+		String[] value = getStringArray(key);
+		KrollDict[] result = new KrollDict[value.length];
+		int index = 0;
+		for (String record : value) {
+			try {
+				result[index++] = new KrollDict(new JSONObject(record));
```

Comment:
From looking at `TiConvert.toStringArray()`, an element in the array can be `null`. So, this will wrongly remove null entries and reduce the size of the array. Perhaps the below would be better?

```java
for (String record : value) {
	KrollDict dictionary = null;
	if (record != null) {
		try {
			dictionary = new KrollDict(new JSONObject(record));
		} catch (Exception e) {
			e.printStackTrace();
			Log.w(TAG, "Unable to parse dictionary.");
		}
	}
	result[index++] = dictionary;
}
```